analyzers/src/SonarAnalyzer.CSharp/Rules/CommentedOutCodeCodeFix.cs

```diff
@@ -0,0 +1,124 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[ExportCodeFixProvider(LanguageNames.CSharp)]
+public sealed class CommentedOutCodeCodeFix : SonarCodeFix
+{
+    public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(CommentedOutCode.DiagnosticId);
+
+    public override FixAllProvider GetFixAllProvider() => null;
+
+    protected override Task RegisterCodeFixesAsync(SyntaxNode root, CodeFixContext context)
+    {
+        var diagnostic = context.Diagnostics.First();
+        var comment = root.FindTrivia(diagnostic.Location.SourceSpan.Start, findInsideTrivia: true);
+
+        if (comment.IsKind(SyntaxKind.SingleLineCommentTrivia) || IsCode(comment))
+        {
+            context.RegisterCodeFix(CodeAction.Create(
+                CommentedOutCode.MessageFormat,
+                c => new Context(comment).ChangeDocument(context.Document, root)),
+                diagnostic);
+        }
+        return Task.CompletedTask;
+    }
+
+    private static bool IsCode(SyntaxTrivia trivia) =>
+        trivia.GetLineNumbers().Count() == 1
+        || Array.TrueForAll(Lines(trivia), IsCode);
+
+    private static bool IsCode(string line)
+    {
+        var trimmed = line.Trim();
+        return trimmed == "/*"
+            || trimmed == "*/"
+            || string.IsNullOrEmpty(trimmed)
+            || CommentedOutCode.IsCode(line);
+    }
+
+    private static string[] Lines(SyntaxTrivia trivia) =>
+        trivia.ToFullString().Split(MetricsBase.LineTerminators, StringSplitOptions.None);
+
+    internal sealed class Context
+    {
+        public Context(SyntaxTrivia comment)
+        {
+            Comment = comment;
+            Leading = Token.LeadingTrivia.ToList();
+            Trailing = Token.TrailingTrivia.ToList();
+
+            IsTrailing = Trailing.Remove(comment);
+
+            if (!IsTrailing)
+            {
+                Leading.Remove(comment);
+            }
+        }
+
+        public bool IsTrailing { get; }
+        public SyntaxTrivia Comment { get; }
+        public List<SyntaxTrivia> Leading { get; }
+        public List<SyntaxTrivia> Trailing { get; }
+        public SyntaxToken Token => Comment.Token;
+
+        public Task<Document> ChangeDocument(Document document, SyntaxNode root) =>
+            Task.FromResult(document.WithSyntaxRoot(root.ReplaceToken(Token, NewToken)));
+
+        public SyntaxToken NewToken =>
```

Comment:
Could this be a method? Something does not feel right with it being a property.
This is the part of your code that does all the clever stuff to figure out how to re-construct the SyntaxTrivia.

Maybe `GenerateNewToken()`?

Reply:
Made it a method (and private).